### ISSUE TITLE
Magic code login method sends email when ACCOUNT_EMAIL_UNKNOWN_ACCOUNTS is False

### ISSUE THREAD

**Issue by jpwarren** [pennersr/django-allauth#3927]
The relatively new Magic Code login mechanism appears to ignore the `ACCOUNT_EMAIL_UNKNOWN_ACCOUNTS` setting and always sends an 'unknown account' email to whatever email address you provide.

There also doesn't appear to be a check to see if signups are disabled, so you could use the magic code mechanism to spam people with invites to sign up to a site, even if signups aren't enabled for that site.

It appears that in `allauth/account/internal/flows/login_by_code.py` the block

```python
    if not users:
        send_unknown_account_mail(request, email)
```

should instead match the code in `allauth/account/forms.py`

```python
    if not self.users:
        if app_settings.EMAIL_UNKNOWN_ACCOUNTS:
            flows.signup.send_unknown_account_mail(request, email)
```

It's unclear to me why you would want to send a magic login link to an email that isn't linked to an account that already exists on the site. There might be a reason I'm unaware of. Shouldn't there be a check to see if signups are permitted first, though? Or is checking the `EMAIL_UNKNOWN_ACCOUNTS` setting sufficient?

**Comment by pennersr** [comment]
> It's unclear to me why you would want to send a magic login link 

It is not doing that... it is sending a "Hey, you don't have an account" mail. This (sending, or pretending to send) is needed to prevent account enumeration.

**Comment by jpwarren** [comment]
Yes, I see. The account enumeration protection I understand.

I was thinking about the abuse case of an attacker using your site to send emails to a bunch of other people, spamming them to sign up to your site. That's not a new problem, though. It's the same challenge as password resets, hence using the same code, which makes sense.

And features like `ACCOUNT_LOGIN_BY_CODE_MAX_ATTEMPTS` help to guard against the abuse case, while preserving the use cases. Fair enough!

**Comment by pennersr** [comment]
Fixed via 62062968

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/3927
